### ISSUE TITLE
Invalid GeoJSON coming from serializing multiple polygons

### ISSUE THREAD

**Issue by lossyrob** [geotrellis/geotrellis#1142]
```scala
val p1: Polygon = ???
val p2: Polygon = ???

Seq(p1, p2).toGeoJson
```
produces something like

```javascript
{"type":"FeatureCollection","features":[{"type":"Polygon","coordinates":[[[-9465961.876490302,29588072.039175928],[-9465961.876490302,29613382.62261249],[-9461069.90681594,29613382.62261249],[-9461069.90681594,29588072.039175928],[-9465961.876490302,29588072.039175928]]]},{"type":"Polygon","coordinates":[[[-9470853.552646479,5713820.738373496],[-9470853.552646479,5723604.677993998],[-9461069.613025976,5723604.677993998],[-9461069.613025976,5713820.738373496],[-9470853.552646479,5713820.738373496]]]}]} 
```

Which doesn't pass geojsonlint.com

**Comment by lossyrob** [comment]
Also, `Seq[Feature[D]]` doesn't have an implicit `toGeoJson`

Also, Seq[Extent] should get an implicit `toGeoJson`

**Comment by lossyrob** [comment]
This implicit:

https://github.com/geotrellis/geotrellis/blob/master/vector/src/main/scala/geotrellis/vector/io/json/package.scala#L20

Should actually look more like

```scala
implicit class FeaturesToGeoJson[D](features: Traversable[Feature[D]]) {
    def toGeoJson(implicit valueJsonWriter: JsonWriter[D]): String = {
      JsonFeatureCollection(features).toJson.compactPrint
    }
  }
```

point is as a user I would like to have a clearer error message when I try to do a `Seq(feature1, feature2).toGeoJson` and the reason it's failing is because the data type doesn't have a JsonWriter found.

**Comment by allixender** [comment]
I guess there needs to be a proper differentiation between Feature (geometry with properties) and Geometry (really just geometry):
Polygon is only a Geometry. The single json Geometries generated are correct, but their Sequence would be a GeometryCollection, so the embedding of a sequence of polygon geometries into a FeatureCollection is wrong. Only PolygonFeature (jtsgeom, data) and a Sequence of those thereof should go into a FeatureCollection, a Sequence of Polygon (jtsgeom) should be embedded into a GeometryCollection.

So apart of the coordinates not plotting nicely, this validates:

```json
{
    "type": "GeometryCollection",
    "geometries": [
        {
        "type": "Polygon",
        "coordinates": [
            [
                [
                    -9465961.876490302,
                    29588072.039175928
                ],
                [
                    -9465961.876490302,
                    29613382.62261249
                ],
                [
                    -9461069.90681594,
                    29613382.62261249
                ],
                [
                    -9461069.90681594,
                    29588072.039175928
                ],
                [
                    -9465961.876490302,
                    29588072.039175928
                ]
            ]
        ]
    }, {
        "type": "Polygon",
        "coordinates": [
            [
                [
                    -9470853.552646479,
                    5713820.738373496
                ],
                [
                    -9470853.552646479,
                    5723604.677993998
                ],
                [
                    -9461069.613025976,
                    5723604.677993998
                ],
                [
                    -9461069.613025976,
                    5713820.738373496
                ],
                [
                    -9470853.552646479,
                    5713820.738373496
                ]
            ]
        ]
    }]
}
```

An Extent can be a Geometry (it has toPolygon or jts.Envelope), but not a Feature, because it doesn't have any other inherent properties.

The logic in fact implies, that a sequence of Polygon (which are only Geometry) should pick up GeometriesToGeoJson implict from package.scala ; so far so good. However, GeometriesToGeoJson refers to JsonFeatureCollection which has an apply for Traversable[Features], but also pure Traversable[Geometries] - that's why it packs normal Geometries into a FeatureCollection -  which is wrong. 

Therefore there needs to be a JsonGeometryCollection or something like that, that handles that different case if desired.

Is it desired? New geotrellis feature handling geometry collection or a fixing this a bug?

**Comment by lossyrob** [comment]
This is as proper of a definition that we'll get: http://geojson.org/geojson-spec.html

Good point about the GeometryCollection...I guess we could right now encode it as a geometry collection. Can we have the example

```scala
val p1: Polygon = ???
val p2: Polygon = ???

val json = Seq(p1, p2).toGeoJson
val polygonsBack = json.parseGeoJson[GeometryCollection].polygons
```

produce the GeoJson for a GeometryCollection containing the polygons?

**Comment by lossyrob** [comment]
val json: String = ???

val p = json.extractGeometries[G <: Geometry]: Seq[G] // May be empty
val p = json.extractFeatures[G <: Geometry, D]: Seq[Feature[G, D]]
val p = json.parseGeoJson[PolygonFeature[Int]]: PolygonFeature[Int] // This expects all the types to line up, throws if not.
val p = json.parseGeojson[JsonFeatureCollection].getAllPolygonFeature[D]().head

p

**Comment by allixender** [comment]
Have to fix the vector-test too

```
- should parse string to points and back again *** FAILED ***
[info]   spray.json.DeserializationException: FeatureCollection expected
[info]   at geotrellis.vector.io.json.FeatureFormats$featureCollectionFormat$.read(FeatureFormats.scala:183)
[info]   at geotrellis.vector.io.json.FeatureFormats$featureCollectionFormat$.read(FeatureFormats.scala:180)
[info]   at spray.json.JsValue.convertTo(JsValue.scala:31)
[info]   at geotrellis.vector.io.json.package$RichString$.parseGeoJson$extension(package.scala:40)
[info]   at geotrellis.vector.io.json.GeoJsonSpec$$anonfun$7.apply$mcV$sp(GeoJsonSpec.scala:61)
[info]   at geotrellis.vector.io.json.GeoJsonSpec$$anonfun$7.apply(GeoJsonSpec.scala:56)
[info]   at geotrellis.vector.io.json.GeoJsonSpec$$anonfun$7.apply(GeoJsonSpec.scala:56)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
```

**Comment by lossyrob** [comment]
Fixed by #1144
